### PR TITLE
fix: tf plan truncated on PR comments and plan_validate_all error logs in github actions

### PR DESCRIPTION
**Pull request by nbugden**
Context: 
The terraform plan generated in GitHub Actions is often truncated when the comment is posted to a PR. This is because the comment is posting the stdout for all environments not just the plan generated by terraform. This PR logs the terraform plan as a separate text file for each environment and then uses those files to create 1 comment per environment on the PR from GitHub Actions.

This also removes the need for #1113 as stdout from the `plan_validate_all` stage is no longer written to a file.

Issue(s) Resolved:
#1109 
#1130

### CODE REVIEW

**Comment by nbugden** [comment]
> @rjerrems @eeaton @gtsorbo can one of you folks take a look. PR needs `/gcbrun` and an approving review.

Nudge, can I get a review. If there is something else required before review, please let me know.

File: build/tf-wrapper.sh
Comment (fmichaelobrien) [inline build/tf-wrapper.sh:142]: These are good - usually the pr lint flags these - I have had to wrap all vars before in other repos

Comment (nbugden) [reply build/tf-wrapper.sh:142]: Thanks
Seems the linter has been `Expected — Waiting for status to be reported` for a number of weeks (possibly waiting for an approval to run). I validated locally using shellcheck. If this is likely to get flagged by the linter I can change it back as it is not functional change that is part of the PR. 

For reference here is the issue flagged by shellcheck: https://www.shellcheck.net/wiki/SC2295

Comment (eeaton) [reply build/tf-wrapper.sh:142]: /gcbrun to get the linter going

**Comment by nbugden** [comment]
@eeaton @apeabody thanks for triggering the checks!
`terraform-example-foundation-int-trigger-HubAndSpoke (cloud-foundation-cicd) ` has failed, but I'm unable to see the logs. Can either of you help me with the error so I can address any changes needed in the PR?

**Comment by apeabody** [comment]
> @eeaton @apeabody thanks for triggering the checks! `terraform-example-foundation-int-trigger-HubAndSpoke (cloud-foundation-cicd) ` has failed, but I'm unable to see the logs. Can either of you help me with the error so I can address any changes needed in the PR?

Might be a flake:
```
Step #30 - "destroy-networks": === NAME  TestNetworks/production
Step #30 - "destroy-networks":     destroy.go:11: 
Step #30 - "destroy-networks":         	Error Trace:	/builder/home/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.11/modules/terraform/destroy.go:11
Step #30 - "destroy-networks":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.12.1/pkg/tft/terraform.go:456
Step #30 - "destroy-networks":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.12.1/pkg/tft/terraform.go:543
Step #30 - "destroy-networks":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.12.1/pkg/tft/terraform.go:556
Step #30 - "destroy-networks":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.12.1/pkg/utils/stages.go:31
Step #30 - "destroy-networks":         	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.12.1/pkg/tft/terraform.go:559
Step #30 - "destroy-networks":         	Error:      	Received unexpected error:
Step #30 - "destroy-networks":         	            	'terraform [destroy -auto-approve -input=false -var access_context_manager_policy_id=727108865530 -var remote_state_bucket=bkt-d5y-b-seed-tfstate-1323 -var ingress_policies=[{"from" = {"sources" = {"access_levels" = ["*"]}, "identity_type" = "ANY_IDENTITY"}, "to" = {"resources" = ["*"], "operations" = {"storage.googleapis.com" = {"methods" = ["google.storage.objects.get", "google.storage.objects.list"]}}}}] -var egress_policies=[{"from" = {"identity_type" = "ANY_IDENTITY"}, "to" = {"resources" = ["*"], "operations" = {"storage.googleapis.com" = {"methods" = ["google.storage.objects.get", "google.storage.objects.list"]}}}}] -var perimeter_additional_members=[] -no-color -lock=false]' unsuccessful after 1 retries
Step #30 - "destroy-networks":         	Test:       	TestNetworks/production
```

**Comment by nbugden** [comment]
Looks like this time both cloud build checks failed. Possibly another flake? I also had to pull in the latest changes. Can you help me with the error so I can address any changes needed in the PR?

**Comment by nbugden** [comment]
Checks have passed. Can a get a review and merge if this is ready to go?

**Comment by nbugden** [comment]
Pulled in the latest from master, can I get a review as well?

**Comment by daniel-cit** [comment]
@nbugden I will try a deploy on github for the review

**Comment by nbugden** [comment]
FYI @daniel-cit I just fixed the named branches in the pipeline triggers. Still had `non-production` from a bad rebase.

**Comment by nbugden** [comment]
@daniel-cit any questions about the PR? Did you get a chance to test it?

**Comment by daniel-cit** [comment]
@nbugden tested
image

**Review by daniel-cit — APPROVED** [review]
LGTM

**Comment by nbugden** [comment]
@daniel-cit one of the integration test pipelines failed

**Comment by daniel-cit** [comment]
@apeabody the build is green :)

**Pull request merged**
